iocore/eventsystem/I_Thread.h

```diff
@@ -63,15 +63,20 @@
 #error "include I_EventSystem.h or P_EventSystem.h"
 #endif
 
+#include <functional>
+
 #include "ts/ink_platform.h"
 #include "ts/ink_thread.h"
 #include "I_ProxyAllocator.h"
 
-class Thread;
 class ProxyMutex;
-typedef void *(*ThreadFunction)(void *arg);
 
-static const int MAX_THREAD_NAME_LENGTH = 16;
+constexpr int MAX_THREAD_NAME_LENGTH = 16;
+
+#define THREADAPI
+
+using THREAD_RETURN_TYPE = void *;
+using ThreadFunction     = std::function<THREAD_RETURN_TYPE THREADAPI()>;
```

Comment:
The return from the thread function isn't used so we can drop `THREAD_RETURN_TYPE` and just do:
```C
using ThreadFunction = std::function<void()>;
```